aea/aea_builder.py

```diff
@@ -640,13 +640,23 @@ def _process_connection_ids(
 
         return sorted_selected_connections_ids
 
-    def build(self, connection_ids: Optional[Collection[PublicId]] = None) -> AEA:
+    def build(
+        self,
+        ledger_api: Optional[LedgerApis] = None,
+        connection_ids: Optional[Collection[PublicId]] = None,
+    ) -> AEA:
         """
         Build the AEA.
 
+        :param ledger_api: the api ledger that we want to use.
         :param connection_ids: select only these connections to run the AEA.
         :return: the AEA object.
         """
+        _ledger_api = (
+            ledger_api
+            if ledger_api is not None
+            else LedgerApis(self.ledger_apis_config, self._default_ledger)
+        )
```

Comment:
We should add some assertions, that the ledger_apis_config and _default_ledger match with what has been set on `ledger_api`...otherwise this could lead to unexpected behaviour

Reply:
If you pass a `ledger_api` in the build function you are skipping this : 
`builder.add_ledger_api_config(FETCHAI, {"network": "testnet"})` 
So I am not sure where should I add the assertion. Possibly before we return the agent ?